hphp/runtime/base/variable-serializer.cpp

```diff
@@ -240,12 +240,15 @@ void VariableSerializer::write(double v) {
     if (!std::isinf(v) && !std::isnan(v)) {
       char *buf;
       vspprintf(&buf, 0, "%.*k", precision, v);
+      if (m_option & k_JSON_PRESERVE_ZERO_FRACTION
+          && strchr(buf, '.') == nullptr) {
+        free(buf);
+        asprintf(&buf, "%.1f", v);
```

Comment:
This is perfect. I start to wear blinders when looking at php-src...
You've done most the work for this diff by now. :D